internal/wasm/store.go

```diff
@@ -246,101 +382,238 @@ func NewStore(ctx context.Context, engine Engine) *Store {
 		Engine:                 engine,
 		maximumFunctionAddress: maximumFunctionAddress,
 		maximumFunctionTypes:   maximumFunctionTypes,
-		maximumGlobals:         maximumGlobals,
 	}
 }
 
+// checkFuncAddrOverflow checks if there would be too many function instances in a store.
+func (s *Store) checkFuncAddrOverflow(newInstanceNum int) error {
+	if len(s.Functions)+newInstanceNum > int(s.maximumFunctionAddress) {
```

Comment:
Is there guarantee that `len(s.Functions)+newInstanceNum` doesn't overflow and doesn't negative value? e.g. it is validated that `newInsanceNum` is small enough beforehand. If not, this add may overflow and the validation can be bypassed, so may be better to replace it with `len(s.Functions) > int(s.maximumFunctionAddress) - newInstaneNum`

Reply:
good point lemme